packages/citrix_adc/data_stream/log/elasticsearch/ingest_pipeline/native.yml

```
@@ -83,7 +85,7 @@ processors:
   - pipeline:
       name: '{{ IngestPipeline "sslvpn_and_aaatm_feature" }}'
       tag: pipeline_sslvpn_and_aaatm_feature
-      if: ctx.citrix?.device_event_class_id != null && (ctx.citrix.device_event_class_id == "SSLVPN" || ctx.citrix.device_event_class_id == "AAATM")
+      if: ctx.citrix?.device_event_class_id != null && ((ctx.citrix.device_event_class_id == "SSLVPN" && ctx.citrix.name.toUpperCase() != "MESSAGE") || ctx.citrix.device_event_class_id == "AAATM")
```

`equalsIgnoreCase` will avoid an allocation. Also good to null check the field.

```suggestion
      if: ctx.citrix?.device_event_class_id != null && ((ctx.citrix.device_event_class_id == "SSLVPN" && ctx.citrix?.name?.equalsIgnoreCase("MESSAGE")) || ctx.citrix.device_event_class_id == "AAATM")
```

---

sweet

```!ctx.citrix?.name?.equalsIgnoreCase("MESSAGE")``` though

---

Oops, good catch!